server/api/controllers/posts.js

```diff
@@ -0,0 +1,37 @@
+const express = require('express');
+const router = express.Router();
+
+const Post = require('../models/posts')
+
+// dogs index route
+router.get('/', async (req, res) => {
+    try {
+        const posts = await Post.all
+        res.json({posts})
+    } catch(err) {
+        res.status(500).json({err})
+    }
+})
+
+// dogs show route
+router.get('/:id', async (req, res) => {
+    try {
+        const post = await Post.findById(parseInt(req.params.id))
+        res.json(post)
+    } catch(err) {
+        res.status(404).json({err})
+    }
+})
+
+// Create post route
+router.post('/', async (req, res) => {
+    try {
```

Comment:
Your app is working, but it's accepting empty strings as inputs. some validation checks might help